libraries/core_libs/consensus/src/pbft/proposed_blocks.cpp

```diff
@@ -27,6 +27,8 @@ bool ProposedBlocks::pushProposedPbftBlock(uint64_t round, const std::shared_ptr
     found_round_it = found_period_it->second.insert({round, {}}).first;
   }
 
+  db_->saveProposedPbftBlock(proposed_block, round);
```

Comment:
Maybe we need to check if this method is called from initialState to not try to save it second time

Reply:
I added a flag to prevent saving data that is already in db